[27기 김성현] [Mission 1-9 완료]

Globalkmaria opened the pull request:
- 가이드에 따른 login , main 페이지 구현 중
- issues 에 있는 mission 에 맞는 구현 사항으로 수정 예정

## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [x] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 가이드에 따른 main, login page 구현
- login page 아이디, 비밀번호 입력창을 사용한 로그인 기능 구현
- login page css, html 완료
- main page 댓글기능 구현 
- main page css, html 완료
<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. 로그인 입력 데이터 유효성 검사 
- 입력창에 변화가 일어날 때마다 버튼 활성화 함수가 실행된다.
- id 1자 이상  +  pw 1자 이상 -> 버튼 활성화됨
- 버튼이 활성화  + 버튼 클릭 또는 input 창에서 엔터시 유효성 검사 ->
  id에 '@'가 포함되지 않으면 '아이디에 '\@'이 포함되어야 합니다.'라는 문구
  pwd가 5자 미만이면 '비밀번호는 5자리 이상이어야 합니다.'라는 문구
  alert 창에 표시됨
- 로그인 되면 main page 로 이동함

2. main 페이지 댓글 기능
- feed 댓글 추가 기능
- input에 따른 게시 버튼 활성화를 버튼 글자색으로 변화해 표현
- enter, 게시 버튼 클릭시 댓글 추가

3. main search input에 따른 input 반응
- search input 창 클릭시 searchTitle 없어짐
- search input에 값이 있는 상태로 blur 되면 안에 적었던 값이 보여짐
- searchTitle에 클릭이벤트를 적용하여 searchTitle 클릭해도 search input 에 focus 가게 함

4. main comment 삭제
- comment hover 시 delete 아이콘 나타나게하기
- commnet마다 comment_$ 로 id 주기, delete 아이콘에 d_$ id 주기
- comments 전체를 감싸는 feedComments에 addEventListener 를 넣어 안에서 일어나는 클릭 이벤트 전체 핸들링하기
- delete 아이콘 클라스를 가진 event.target 일 때 delete 실행하기
- delete 아이콘 클릭 시 해당 id 를 받아와 그 id를 가진 comment 지우기

5. comment 좋아요 기능
- 2개의 하트를 겹치게 하여 앞의 하트의 opacity를 조절해 하트 좋아요 보여주기
- toggle class로 opacity를 적용
- feedComments.addEventListener에 적용해 한꺼번에 click 이벤트 적용

6. media 설정 width: 900px 이하 일 때
- main 오른쪽 부분 사라지게 하기
- feed 부분 중앙 배치

7. user name 검색 기능
- 글자가 입력될 때마다 검색 결과 창 새로 만들어짐
- input 값이 없거나 일치하는 유저가 없을 때 '결과 없음'이라는 문구 보임
- 검색이 끝나고 input blur로 만들면 창 없어짐

8. nav 프로필 사진 클릭 시 메뉴 박스 생성
- 프로필 사진 클릭 시 메뉴 박스 생성
- 프로필 사진이나 메뉴 박스 내 클릭 시 메뉴 계속 보임
- 프로필, 메뉴 박스 이외 바깥 부분 클릭 시 메뉴 박스 사라짐
<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게 된 부분, 어려웠던 점 등을 작성합니다.)
- search input의 더 부드러운 사용을 위해 search text를 누르더라도 input에 focus 가 되도록 만들면서
부드러운 사용감에 대해 더 이해할 수 있었습니다.
- background를 content-background 배경 색만을 위해 사용했는데
story 둥근 띠를 만들면서 background에 padding, border의 색을 지정해 줄 수 있다는 것을 알게 되었습니다.
- id 에는 space를 줄 수 없다는 것을 알게 되었습니다. 대신 '_' 쓰기.
- opacity를 toggle 해줄 때는 직접 해주는 것이 아닌 classList.toggle 을 이용해서 변경해 주기
<br />
- 성장하고 있는가에 대해 고민을 많이 하고 있엇는데 예전에 짰던 outside click 코드보다  매우 간결하고 코드 어디에서든 간단하게 적용할 수 있는 코드를 작성해서 성장했구나를 느껴 정말 기분이 좋은 구현이었다!!

## :: 기타 질문 및 특이 사항

dipsiiiiiiiiii commented on js/login.js:
```diff
@@ -0,0 +1,11 @@
+let user = document.getElementById('user');
+let pswd = document.getElementById('pswd');
+let bttn = document.getElementById('login');
```
let을 사용할 때는 이 변수의 값을 새로 할당하는 경우에 사용합니다!
지금은 변수 선언을 하고 난 후에 값이 변하지 않으므로 const 사용해주세요 :)

그리고 변수명만 봐도 어떤 태그인지 알 수 있게 
```userInput``` 이렇게 적어주시면 좋을 것 같네요!

dipsiiiiiiiiii replied:
그리고 id를 사용하고 계신데 id는 같이 협업할 때 겹칠 가능성이 있어서 안 쓰는 게 좋습니다!
지금은 혼자 프로젝트를 해서 상관없지만 만약에 프로젝트 사이즈가 커진다면
자기가 쓴 id도 기억 못하고 한 번 더 써서 오류가 날 수 있습니다!
그래서 클래스를 사용해주세요 :)

lemongucci replied:
>>다만 작업하고 있는 파일 위치를 students 폴더 아래에 13th 폴더 아래로 해주세요!
가이드 영상 보시면 나올겁니다!!
re: 관희님...ㅜㅜㅠ 아무리 해도 저는 가이드대로 잘 한 것 같은데, 파일 위치를 어떻게 변경해야 하는 지 모르겠습니다.

dipsiiiiiiiiii commented on js/login.js:
```diff
@@ -0,0 +1,11 @@
+let bttn = document.getElementById('login');
+
+pswd.addEventListener('input', function(e){
+  if(e !== '') {
```
이 조건의 의미는 어떤건가요?

lemongucci replied:
input event가 false 같지 않으면, 한 마디로 input event가 발생하면, if statement의 코드가 실행됩니다.

dipsiiiiiiiiii commented on js/main.js:
```diff
@@ -0,0 +1,27 @@
+let pstdTime = document.getElementById('pstdTime');
+let social = document.getElementById('social');
+let cmnt = document.getElementById('newCmnt');
+let bttn = document.getElementById('pstBttn');
```
여기도 위 리뷰 반영해서 수정해주세요 :)

lemongucci replied:
1. 관희님 그런데 혹시 textarea는 css로 가능한 스타일링이 한정적인가요? 
replaced element라고 css로 지정 가능한 것이 위치 정도라고 하는데, 
2. 그렇다면 textarea 보다 input으로 댓글창을 만드는 게 더 좋을까요?
그런데 실제 인스타그램에 들어가보면 댓글창이 textarea로 되어있어서, 도대체 어떻게 스타일링을 한 것인지 잘 모르겠습니다.
3. textarea 스타일링을 더 잘 할 수 있는 팁이나 방법이 있나요?

dipsiiiiiiiiii commented on js/main.js:
```diff
@@ -0,0 +1,27 @@
+let x = 'silverglasses';
+let y = cmnt.value;
+let z;
+bttn.disabled = true;
+bttn.style.opacity = '0.3';
```
초기값을 주려고 이렇게 하신 것 같은데 그러면 css 파일에 바로 넣어주시는 게 좋을 것 같아요!

lemongucci replied:
초기값은 왜 css파일에 넣어서 관리하는 게 더 편하죠? 위 아래에 초기값, 변경값이 같이 있으면 일일히 기억하지 않아도 되서 더 편하지 않나요?
(물론 관희님 말씀대로 css에 넣겠습니다!)

dipsiiiiiiiiii commented on login.html:
```diff
@@ -0,0 +1,22 @@
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width">
+    <title>repl.it</title>
```
타이틀도 바꿔보시면 좋을 것 같아요!

dipsiiiiiiiiii commented on login.html:
```diff
@@ -0,0 +1,22 @@
+     <link href="style/login.css" rel="stylesheet" type="text/css" />
+  </head>
+  <body>
+    <div>
+      <section>
+        <img src='img/logo_text.png' alt='logo'>
+        <input id='user' type="text" placeholder="Phone number, username, or email">
+        <input id='pswd' type="password" placeholder="Password">
+        <button id='login' type="submit">Log In</button>
+      </section>
+      <p>Forgot password?</p>
+    </div>
```
굿! 너무 잘 하셨어요! 이렇게 깔끔한 태그는 너무 오랜만이네요!

lemongucci replied:
감사합니다!!! 제가 동기분들한테 관희님 거의 칭찬봇이어서 리뷰 받으면 굉장히 기운이 난다고!!! 북돋아 주셔서 감사합니다. 기분이 좋네여...ㅎㅎㅎ

dipsiiiiiiiiii commented on main.html:
```diff
@@ -0,0 +1,140 @@
+     <link href="style/main.css" rel="stylesheet" type="text/css" />
+  </head>
+  <body>
+    <nav>
```
nav 태그 너무 좋아요!

dipsiiiiiiiiii commented on main.html:
```diff
@@ -0,0 +1,140 @@
+        <img id='imgLogo' src='https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/logo.png' alt='Image Logo'>
+        <img id='imgText' src='img/logo_text.png' alt='Image Text'>
+      </span>
+      <span>
+        <input type="text" placeholder="Search">
+      </span>
```
span 태그로 안 감싸도 될 것 같아요!

lemongucci replied:
span이 굳이 없어도 되지만, css에서 스타일링할 때, nth-child로 구분하는 게 편해서 그냥 남겨두어도 될까요?

dipsiiiiiiiiii commented on main.html:
```diff
@@ -0,0 +1,140 @@
+      </span>
+    </nav>
+
+    <main>
```
main태그도 너무 좋네요!

dipsiiiiiiiiii commented on main.html:
```diff
@@ -0,0 +1,140 @@
+    </nav>
+
+    <main>
+      <div class='feeds'>
+        <article>
```
div태그의 클래스를 article에서 사용하면 div는 필요 없을 것 같아요 :)

dipsiiiiiiiiii commented on main.html:
```diff
@@ -0,0 +1,140 @@
+            <section class="prfl">
+              <img id='prflPhoto'src="https://scontent-gmp1-1.cdninstagram.com/v/t51.2885-19/s150x150/71533411_449636125903525_7464596574253875200_n.jpg?_nc_ht=scontent-gmp1-1.cdninstagram.com&_nc_ohc=Izru35a3h2sAX8nevRu&oh=b911dfd175a23d0195d6d8fdf1bd2b36&oe=5F91D92B" alt="Profile photo">
+              <div>
+                <p>jennierubyjane</p>
+                <p>Black Pink</p>
```
Semantic하게 써주셔서 너무 좋네요!!

dipsiiiiiiiiii commented on main.html:
```diff
@@ -0,0 +1,140 @@
+            </section>
+            <img id="more" src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/more.png" alt="Three dots" width="16" height="16">
+          </div>
+          <div class='photo'>
+            <img src="https://scontent-gmp1-1.cdninstagram.com/v/t51.2885-15/sh0.08/e35/s750x750/119945872_660222924899582_7313075262915608996_n.jpg?_nc_ht=scontent-gmp1-1.cdninstagram.com&_nc_cat=1&_nc_ohc=I1HFmacSFvYAX_6IbAb&oh=058d62956447c7f7c8b15a1b13abfe1c&oe=5F9435B2" alt="Feed" width="614px" height="614px">
+          </div>
```
div태그로 안감싸도 괜찮을 것 같아요!

lemongucci replied:
네 div 치웠습니다.

dipsiiiiiiiiii commented on main.html:
```diff
@@ -0,0 +1,140 @@
+          </div>
+          <div id='social'>
+            <section>
+              <div id='fdbLeft'>
```
fdb가 무엇인가요!? 좀 더 명확하게 써주시면 좋을 것 같아요 :)

lemongucci replied:
feed bar의 left side를 말하고 싶었어요 ㅎㅎㅎ

dipsiiiiiiiiii commented on main.html:
```diff
@@ -0,0 +1,140 @@
+          <div id='social'>
+            <section>
+              <div id='fdbLeft'>
+                <img src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/heart.png" alt="Like" width="24px" height="24px">
+                <img src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/comment.png" alt="Comment" width="24px" height="24px">
+                <img src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/share.png" alt="Share" width="24px" height="24px">
```
alt 속성 너무 좋네요!
다만, alt 속성이 이미지에 대해서 설명해 주는 부분이어서 src 속성보다 중요합니다!
alt를 더 앞에 적어주세요 :)

lemongucci replied:
예 알겠습니다

dipsiiiiiiiiii commented on main.html:
```diff
@@ -0,0 +1,140 @@
+        </article>
+      </div>
+      <div class='main-right'>
+        <div>
```
이 div는 없어도 될 것 같네요 :)

lemongucci replied:
70번 라인 div 말씀하시는 거죠?
css에서 flex box 효과 주려고 parent필요해서 div를 사용했습니다. 안에 img랑 section으로 자식 요소 들어가 있어요.

dipsiiiiiiiiii reviewed (commented):
너무 잘 짜주셨네요!
Semantic 태그를 사용하시려는 고민이 보여서 너무 좋았습니다!
여태까지 본 코드 중에 제일 잘 고민해서 짜주신 분 같아서 리뷰 할 때도 너무 좋았네요!

다만 작업하고 있는 파일 위치를 students 폴더 아래에 13th 폴더 아래로 해주세요!
가이드 영상 보시면 나올겁니다!!

고생 너무 많이 하셨어요!

lemongucci commented:
> 너무 잘 짜주셨네요!
> Semantic 태그를 사용하시려는 고민이 보여서 너무 좋았습니다!
> 여태까지 본 코드 중에 제일 잘 고민해서 짜주신 분 같아서 리뷰 할 때도 너무 좋았네요!
> 
> 다만 작업하고 있는 파일 위치를 students 폴더 아래에 13th 폴더 아래로 해주세요!
> 가이드 영상 보시면 나올겁니다!!
> 
> 고생 너무 많이 하셨어요!

Globalkmaria committed:
make sunghyun folder , add all files
- login , main

Globalkmaria committed:
add 삼항연산자 in login.js

Globalkmaria committed:
\[add\]
add feed profile description
modify root gray color
modify navWrapper, main -  max-width
modify search box
modify copyright width

Globalkmaria committed:
[add]id, pw validation

Globalkmaria committed:
[add]commit delete 기능 추가

Globalkmaria committed:
[add]comment like 기능 추가

Globalkmaria committed:
[Refactor] css refactoring

Globalkmaria committed:
[Refactor] js 변수이름 refactoring

Globalkmaria committed:
[Refactor] semantic Tag,  html img alt 추가

Globalkmaria committed:
[Modify] feed 이미지 화면크기에 따라 크기 변경되게하기

Globalkmaria committed:
[Fix] js refactoring하며 작동이 안된 searchTitle handler 재 구현

Globalkmaria committed:
[Refactor] login validation check method 개선

Globalkmaria committed:
[Add] max-width: 900px -> main-right 사라지게하기

Globalkmaria committed:
[ADD] username 검색기능 추가

Globalkmaria committed:
[Modify] 검색결과가 없을때에도 검색결과없음문구를 보려주도록 만듦

Globalkmaria committed:
[Add] login page -> login 성공시 main page 로 이동

Globalkmaria committed:
[Add] main page 내용없는 댓글 추가되지않는기능 추가

Globalkmaria committed:
[Modify] small changes

Globalkmaria committed:
[Add] 프로필 사진 클릭시 메뉴 박스 생성

Globalkmaria committed:
[Modify] outside click event modify
